### ISSUE TITLE
[blockly] Latest UoM QTY blocks with variables no longer work

### ISSUE THREAD

**Issue by Mark-VG** [openhab/openhab-webui#1907]
## The problem
EDIT: Confirmed same issue on 4.0.0.M3

Running on openHAB 4.0.0 Build #3470

The latest UoM blocks produce incorrect code when working with variables.

![image](https://github.com/openhab/openhab-webui/assets/84638697/99c0ea87-cf6c-410b-a9b2-eaa77f89f73e)
 
Produces the following code

```
var Test1;


Test1 = Quantity('10 V');
console.debug((Quantity('100 V').add()));
console.debug((Quantity('100 V').add(Quantity('10 V'))));
```

As can be seen from this example the first formula has the variable missing and produces an invalid function.
```
console.debug((Quantity('100 V').add(***Missing***)));
```
As far as I am aware the produced code with the vriable should be:
```
console.debug((Quantity('100 V').add(Quantity(Test1))));
```
## Expected behavior

Produced code from Blockly to be correct and functional

## Steps to reproduce

Create Blockly Script as per above screen shot and check resultant code. Sample rule included below.

## Your environment

```
Configuration folder     C:\openHAB\conf
User data folder     C:\openHAB\userdata
Logs folder     C:\openHAB\userdata\logs
Operating system     Windows 10/10.0 (amd64)
Java Runtime     Azul Systems, Inc. 17.0.7 (Zulu17.42+19-CA)
openHAB 4.0.0 Build #3470
```
Test Rule Code

```

Rules
 Loading...
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Title of the rule
status badge
Tags, Schedule, Scene...
Description of the rule
Test UoM
1
configuration: {}
2
triggers:
3
  - id: "1"
4
    configuration:
5
      itemName: Testing_Switch
6
    type: core.ItemStateChangeTrigger
7
conditions: []
8
actions:
9
  - inputs: {}
10
    id: "2"
11
    configuration:
12
      blockSource: <xml
13
        xmlns="https://developers.google.com/blockly/xml"><variables><variable
14
        id="CT7z)^UGiUA(a76IsM:$">Test1</variable></variables><block
15
        type="variables_set" id=":iV4T[iuFW+cy6KC([h?" x="45" y="49"><field
16
        name="VAR" id="CT7z)^UGiUA(a76IsM:$">Test1</field><value
17
        name="VALUE"><block type="oh_quantity" id="xPzls{{)@9)7x*}c|mRv"><value
18
        name="quantity"><shadow type="text" id="AuQ73xk/%R6$@X.+({qU"><field
19
        name="TEXT">10 W</field></shadow><block type="text"
20
        id="D+[g*~hRODXW5f7DXLiK"><field name="TEXT">10
21
        V</field></block></value></block></value><next><block type="oh_log"
22
        id="mEiA99CDb5q1T;vbP;vy"><field name="severity">debug</field><value
23
        name="message"><shadow type="text" id="Q(`Wk5N[n~sZc@]d#(%7"><field
24
        name="TEXT">abc</field></shadow><block type="oh_quantity_arithmetic"
25
        id="qS`I@wv(9,`Ty#RrbGrL"><field name="operand">add</field><value
26
        name="first"><shadow type="oh_quantity"
27
        id="%D`X+rWrZ7K5/3V.$`u8"></shadow><block type="oh_quantity"
28
        id="[}bkJgD0298LLTrufH=k"><value name="quantity"><shadow type="text"
29
        id="Rg*ihu5k76vFl5Vz[1_3"><field name="TEXT">10 W</field></shadow><block
30
        type="text" id="MDRHDmjqlN*np=qhk/-K"><field name="TEXT">100
31
        V</field></block></value></block></value><value name="second"><shadow
32
        type="oh_quantity" id="N]Ud`r+6Qoqcf6=b)TpZ"></shadow><block
33
        type="variables_get" id=";GyI7hnYXW^u]]S+M:6K"><field name="VAR"
34
        id="CT7z)^UGiUA(a76IsM:$">Test1</field></block></value></block></value><next><block
35
        type="oh_log" id=":UXrU[e2pMocUi0=LE`k"><field
36
        name="severity">debug</field><value name="message"><shadow type="text"
37
        id="Q(`Wk5N[n~sZc@]d#(%7"><field name="TEXT">abc</field></shadow><block
38
        type="oh_quantity_arithmetic" id="`Lw6!Ya.;:?9YtX!EbS["><field
39
        name="operand">add</field><value name="first"><shadow type="oh_quantity"
40
        id="%D`X+rWrZ7K5/3V.$`u8"></shadow><block type="oh_quantity"
41
        id="Gsi=M:#=o:QZ4E+%@PA["><value name="quantity"><shadow type="text"
42
        id="Rg*ihu5k76vFl5Vz[1_3"><field name="TEXT">10 W</field></shadow><block
43
        type="text" id=",^(%:k7oOoNNDjpz2yR]"><field name="TEXT">100
44
        V</field></block></value></block></value><value name="second"><shadow
45
        type="oh_quantity" id="N]Ud`r+6Qoqcf6=b)TpZ"></shadow><block
46
        type="oh_quantity" id="U?jd4P8Jd0^wa=c4s!N7"><value
47
        name="quantity"><shadow type="text" id="AuQ73xk/%R6$@X.+({qU"><field
48
        name="TEXT">10 W</field></shadow><block type="text"
49
        id="uRHpZ@.NQ6WnBg+vPL7u"><field name="TEXT">10
50
        V</field></block></value></block></value></block></value></block></next></block></next></block></xml>
51
      type: application/javascript
52
      script: |
53
        var Test1;
54
​
55
​
56
        Test1 = Quantity('10 V');
57
        console.debug((Quantity('100 V').add()));
58
        console.debug((Quantity('100 V').add(Quantity('10 V'))));
59
    type: script.ScriptAction
60
​```

Community Forum Topic for further information:

https://community.openhab.org/t/oh4-blockly-persistence-calculation-error/146841/13

**Comment by stefan-hoehn** [comment]
see discussion here

https://community.openhab.org/t/oh4-blockly-persistence-calculation-error/146841/19

**Comment by florian-h05** [comment]
Fixed by #1911.
